### PR TITLE
[functionapp] add support for runtime versions

### PR DESCRIPTION
**Pull request by ankitkumarr**
Fixes  https://github.com/Azure/azure-cli/issues/10766 (https://github.com/Azure/Azure-Functions/issues/1355)
Fixes https://github.com/Azure/azure-cli/issues/10549 (https://github.com/Azure/Azure-Functions/issues/1341)

cc: @Hazhzeng @divyagandhii

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py [hunk 1]
```diff
@@ -1100,8 +1100,8 @@ def test_functionapp_consumption_e2e(self, resource_group, storage_account):
             JMESPathCheck('kind', 'functionapp'),
             JMESPathCheck('name', functionapp_name)
         ])
-        self.cmd('functionapp update -g {} -n {} --set clientAffinityEnabled=true'.format(resource_group, functionapp_name), checks=[
-            self.check('clientAffinityEnabled', True)
+        self.cmd('functionapp update -g {} -n {} --set clientAffinityEnabled=true'.format(resource_group, functionapp_name),
+            checks=[self.check('clientAffinityEnabled', True)
```
Comment (ankitkumarr) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py]: need to reset this

**Comment by qwordy** [comment]
Please fix errors in CI.
Be careful with runtime, runtime.lower(), runtime.upper().
Thanks.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 2]
```diff
@@ -2336,6 +2349,23 @@ def create_function(cmd, resource_group_name, name, storage_account, plan=None,
...
+    if runtime_version is None:
+        runtime_version = RUNTIME_TO_DEFAULT_VERSION[runtime]
+    if is_consumption:
+        return '{}|{}'.format(runtime.upper(), runtime_version)
+    # App service or Elastic Premium
+    return _format_fx_version(RUNTIME_TO_IMAGE_FUNCTIONAPP[runtime][runtime_version])
+
+
+def _get_website_node_version_functionapp(runtime, runtime_version):
+    if runtime.lower() != 'node':
```
Comment (ankitkumarr) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py]: check for not None

**Comment by ankitkumarr** [comment]
CI is stuck, closing and re-openening

**Comment by yonzhan** [comment]
@qwordy please help with this PR.

**Pull request merged**